### DISCUSSION TITLE
Issue with Torchvision when using Opensoundscape on Jupyter Lab

### DISCUSSION THREAD

**Discussion by francesco99bini** [kitzeslab/opensoundscape#1050] [Q&A]
Hello!

I am new here and I am still quite a beginner with Python. I need to use a code provided by a paper for my MSc thesis, and the authors recommend using Python 3.7 and Opensoundscape 0.7.

I previously ran into errors (regarding Ray) using that version of Python, so I have now tried to use more recent ones.
I followed the instructions in the installation section for Windows of the Opensoundscape documentation, and succesfully started Jupyter Lab. I created an environment with Python 3.10, in which I installed both Opensoundscape 0.7 and 0.10.1 (as I have seen in the instructions).

However, when using either one I run in this error:

![opensoundscape_error2](https://github.com/user-attachments/assets/6ee1aebd-b41f-47af-9279-65b82a680755)

I tried to use both versions of Opensoundscape in another environment with Python 3.9, but I still got the same error.

Does anyone happen to have an idea for a solution?

Thank you very much in advance for your time!

Francesco Bini

**Comment by sammlapp** [comment]
Hi Francesco,
It seems like this error occurs because you have an old version of OpenSoundscape with a new version of the torch package. This suggests you need to either use a newer version of opensoundscape or an older version of torch. I don’t think you would get this error with a newer version of opensoundscape, because more recent version of opensoundscape use the pytorch implementations of resnet rather than a local opensoundscape/torch/architectures/resnet.py module.

My advice is to stick with the newer package versions if possible, and double check that the environment you’re using has opensoundscape 0.10.1 installed in it. If you get a different error when using opensoundscape 0.10.1 and need help, let me know.

Cheers,
Sam

From: francesco99bini ***@***.***>
Date: Wednesday, September 4, 2024 at 12:50 PM
To: kitzeslab/opensoundscape ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [kitzeslab/opensoundscape] Issue with Torchvision when using Opensoundscape on Jupyter Lab (Discussion #1050)

Hello!

I am new here and I am still quite a beginner with Python. I need to use a code provided by a paper for my MSc thesis, and the authors recommend using Python 3.7 and Opensoundscape 0.7.

I previously ran into errors (regarding Ray) using that version of Python, so I have now tried to use more recent ones.
I followed the instructions in the installation section for Windows of the Opensoundscape documentation, and succesfully started Jupyter Lab. I created an environment with Python 3.10, in which I installed both Opensoundscape 0.7 and 0.10.1 (as I have seen in the instructions).

However, when using either one I run in this error:

opensoundscape_error2.png (view on web)<https://github.com/user-attachments/assets/6ee1aebd-b41f-47af-9279-65b82a680755>

I tried to use both versions of Opensoundscape in another environment with Python 3.9, but I still got the same error.

Does anyone happen to have an idea for a solution?

Thank you very much in advance for your time!

Francesco Bini

**Reply by francesco99bini** [reply]
Thank you very much for the quick reply Sam.

I installed the 0.10.1 version and made sure I imported it in the notebook I am using (should I worry about that warning message?).

![opensoundscape_error3](https://github.com/user-attachments/assets/1f2270c1-028a-4d60-963c-b9df82a535d6)

However at the end of the code I get this error now:

![opensoundscape_error4](https://github.com/user-attachments/assets/c11c707a-d346-486c-9bb3-a26efb2d6832)

I tried importing torch and it did it succesfully, but then when I run that line I still get the error shown in the second picture (even after restarting the kernel).

Thanks again!

Best,

Francesco

**Comment by sammlapp** [comment]
Ok, this error does stem from a difference in the opensoundscape version between the original user and your environment. If you want to use the script without modification you should install the version of opensoundscape they originally used and also try to march other package versions. The ideal scenario would be if the original developer could share an environment .yml file listing all of the package versions in their environment (if you use conda: conda env export -f descriptive_name.yml) which will allow you to recreate the same environment so that the script runs smoothly. If this is not available it might take some experimentation to find a working set of packages but I can share a .yml with a valid combination of package versions for opensoundscape 0.7.0.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by francesco99bini** [reply]
I see now, thanks a lot!

If it's really possible for you to share that combination of packages it would be great, otherwise if you preferred me to first contact the authors of that paper I can definitely do that and get back to you in case that option doesn't work

Francesco

**Reply by sammlapp** [reply]
[opso070.yml.zip](https://github.com/user-attachments/files/16895267/opso070.yml.zip)
here is one example of a valid OpenSoundscape environment. You can download, unzip to get the .yml file, and create an environment with `conda create -n myopso070env -f /path/to/opso070.yml`

However, I do recommend you ask for the yml file listing the exact packages used by the original script authors, as that will ensure consistent behavior.

**Reply by francesco99bini** [reply]
That's perfect, thank you very much again for your time and help.

Regards,

Francesco
